bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2439,6 +2439,30 @@ void RewriteInstance::readDynamicRelocations(const SectionRef &Section,
     if (Symbol)
       SymbolIndex[Symbol] = getRelocationSymbol(InputFile, Rel);
 
+    const uint64_t ReferencedAddress = SymbolAddress + Addend;
+    BinaryFunction *Func =
+        BC->getBinaryFunctionContainingAddress(ReferencedAddress);
+
+    if (Relocation::isRelative(RType) && SymbolAddress == 0) {
+      if (Func) {
+        if (!Func->isInConstantIsland(ReferencedAddress)) {
+          if (const uint64_t ReferenceOffset =
+                  ReferencedAddress - Func->getAddress()) {
+            Func->addEntryPointAtOffset(ReferenceOffset);
+          }
+        } else {
+          BC->errs() << "BOLT-ERROR: referenced address at 0x"
+                     << ReferencedAddress
```

Comment:
```suggestion
                     << Twine::utohexstr(ReferencedAddress)
```

Reply:
>please address the comment regarding the hex print before committing

I apologise, I thought I addressed that in my previous commit. It is in hex print now. Thanks for the review!